app/scripts/controllers/console.js

```diff
@@ -67,7 +67,9 @@ angular.module('console', ['sql', 'datatypechecks'])
           if ($scope.useLocalStorage) {
             localStorage.setItem('crate.console.query_list', JSON.stringify(self.recentQueries));
           }
-          inputScope.recentCursor = -1;
+          if (inputScope) {
```

Comment:
what if `inputScope` is `0`?

```js
if (inputScope !== null) {
    inputScope.recentCursor = -1;
}
```

Reply:
the issue is when `inputScope` is null, i get a referenceError.
in line 359 we set it to `$console.setInputScope($scope);` I'm not sure when it can be 0